{"name": "测试步骤1", "status": "broken", "statusDetails": {"message": "TypeError: a bytes-like object is required, not 'NoneType'", "trace": "self = <src.case.test_08allure测试报告介绍.TestResult object at 0x10e88c050>\n\n    @allure.title(\"测试步骤1\")\n    def test_01(self):\n        with allure.step(\"打开浏览器\"):\n            driver=Pyselenium()\n    \n        with allure.step(\"打开百度\") :\n            driver.get(\"https://www.baidu.com\")\n    \n        with allure.step(\"找到wd的元素\") :\n            driver.send_keys((\"id\",\"kw\"),\"python\")\n            driver.click((\"id\",\"su\"))\n    \n        with allure.step(\"结果截图\"):\n>           allure.attach(name=\"执行结果\",\n                          body=driver.get_screenshot_as_png())\n\nsrc/case/test_08allure测试报告介绍.py:59: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/allure_commons/_allure.py:200: in __call__\n    plugin_manager.hook.attach_data(body=body, name=name, attachment_type=attachment_type, extension=extension)\n/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/pluggy/_hooks.py:513: in __call__\n    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)\n/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/pluggy/_manager.py:120: in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/allure_pytest/listener.py:252: in attach_data\n    self.allure_logger.attach_data(uuid4(), body, name=name, attachment_type=attachment_type, extension=extension)\n/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/allure_commons/reporter.py:165: in attach_data\n    plugin_manager.hook.report_attached_data(body=body, file_name=file_name)\n/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/pluggy/_hooks.py:513: in __call__\n    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)\n/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/pluggy/_manager.py:120: in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <allure_commons.logger.AllureFileLogger object at 0x10e878440>, body = None\nfile_name = 'e5f97426-58ad-4330-8313-2fcc7fc39b87-attachment.attach'\n\n    @hookimpl\n    def report_attached_data(self, body, file_name):\n        destination = self._report_dir / file_name\n        with open(destination, 'wb') as attached_file:\n            if isinstance(body, str):\n                attached_file.write(body.encode('utf-8'))\n            else:\n>               attached_file.write(body)\nE               TypeError: a bytes-like object is required, not 'NoneType'\n\n/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/allure_commons/logger.py:48: TypeError"}, "steps": [{"name": "打开浏览器", "status": "passed", "start": 1753179963811, "stop": 1753179966212}, {"name": "打开百度", "status": "passed", "start": 1753179966212, "stop": 1753179968258}, {"name": "找到wd的元素", "status": "passed", "start": 1753179968258, "stop": 1753179968836}, {"name": "结果截图", "status": "broken", "statusDetails": {"message": "TypeError: a bytes-like object is required, not 'NoneType'\n", "trace": "  File \"/Users/admin/Documents/pythonproject/pythonjichu4/src/case/test_08allure测试报告介绍.py\", line 59, in test_01\n    allure.attach(name=\"执行结果\",\n    ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^\n                  body=driver.get_screenshot_as_png())\n                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/allure_commons/_allure.py\", line 200, in __call__\n    plugin_manager.hook.attach_data(body=body, name=name, attachment_type=attachment_type, extension=extension)\n    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/pluggy/_hooks.py\", line 513, in __call__\n    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)\n           ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/pluggy/_manager.py\", line 120, in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n           ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/pluggy/_callers.py\", line 139, in _multicall\n    raise exception.with_traceback(exception.__traceback__)\n  File \"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/pluggy/_callers.py\", line 103, in _multicall\n    res = hook_impl.function(*args)\n  File \"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/allure_pytest/listener.py\", line 252, in attach_data\n    self.allure_logger.attach_data(uuid4(), body, name=name, attachment_type=attachment_type, extension=extension)\n    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/allure_commons/reporter.py\", line 165, in attach_data\n    plugin_manager.hook.report_attached_data(body=body, file_name=file_name)\n    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/pluggy/_hooks.py\", line 513, in __call__\n    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)\n           ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/pluggy/_manager.py\", line 120, in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n           ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/pluggy/_callers.py\", line 139, in _multicall\n    raise exception.with_traceback(exception.__traceback__)\n  File \"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/pluggy/_callers.py\", line 103, in _multicall\n    res = hook_impl.function(*args)\n  File \"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/allure_commons/logger.py\", line 48, in report_attached_data\n    attached_file.write(body)\n    ~~~~~~~~~~~~~~~~~~~^^^^^^\n"}, "attachments": [{"name": "执行结果", "source": "e5f97426-58ad-4330-8313-2fcc7fc39b87-attachment.attach"}], "start": 1753179968836, "stop": 1753179969101}], "attachments": [{"name": "stderr", "source": "be312249-346c-4df2-82b2-210c95ac686d-attachment.txt", "type": "text/plain"}], "start": 1753179963811, "stop": 1753179969113, "uuid": "bb68edc8-5ddf-4614-b085-69f54899e2d5", "historyId": "defd2e42b50608898efe0f3cafaddb87", "testCaseId": "defd2e42b50608898efe0f3cafaddb87", "fullName": "src.case.test_08allure测试报告介绍.TestResult#test_01", "labels": [{"name": "parentSuite", "value": "src.case"}, {"name": "suite", "value": "test_08allure测试报告介绍"}, {"name": "subSuite", "value": "TestResult"}, {"name": "host", "value": "admindeMacBook-Pro.local"}, {"name": "thread", "value": "60315-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "src.case.test_08allure测试报告介绍"}]}